packages/cli/src/services/experiment/ExperimentFlagComparator.ts

```diff
@@ -0,0 +1,102 @@
+import {
+  addChange,
+  addChangeExplicit,
+  ComparisonResult,
+  ValueChangeMap
+} from "../../comparison/ComparisonResult";
+import { ExperimentFlagModel, VariantModel } from "./models";
+import { groupBy, isEmpty } from "lodash";
+import { ExperimentConfigModel } from "../../config/ExperimentsConfig";
+import { sortAlphabetically } from "../../generators/util/sorting";
+
+interface ExperimentFlagVersionComparison {
+  result: ComparisonResult;
+  origin: ExperimentFlagModel;
+  target: ExperimentFlagModel;
+}
+
+interface ExperimentConfigFlagVersionComparison {
+  result: ComparisonResult;
+  origin: ExperimentConfigModel;
+  target: ExperimentConfigModel;
+  changes: ValueChangeMap;
+}
+
+export class ExperimentFlagComparator {
+  compare(origin: ExperimentFlagModel, target: ExperimentFlagModel): ExperimentFlagVersionComparison {
+    let result = ComparisonResult.NoChanges;
+    if (origin.key != target.key) {
+      result = ComparisonResult.Updated;
+    }
+
+    if (origin.name != target.name) {
+      result = ComparisonResult.Updated;
+    }
+
+    if (origin.variants.length != target.variants.length) {
+      result = ComparisonResult.Updated;
+    } else  {
+      const targetVariantMap = groupBy(target.variants, 'key');
+      origin.variants.forEach(ov => {
+        const targetVariant: VariantModel = targetVariantMap[ov.key]?.[0];
+        if (!targetVariant) {
+          result = ComparisonResult.Updated;
+        }
+      })
+    }
+
+    return {
+      result,
+      origin,
+      target,
+    }
+  }
+
+  compare2(origin: ExperimentConfigModel, target: ExperimentConfigModel): ExperimentConfigFlagVersionComparison {
```

Comment:
Minor, ditto.

Reply:
Same as above. Will clean up after merging in latest from main branch.